lib/iam_ssoe_oauth/session_manager.rb

```diff
@@ -44,24 +44,39 @@ def logout
     def create_user_session
       Rails.logger.info('IAMUser create_user_session: start')
 
-      iam_profile = iam_ssoe_service.post_introspect(@access_token)
-      Rails.logger.info('IAMUser create_user_session: introspect succeeded')
-      iam_profile = add_missing_fediamassur_level(iam_profile)
+      StatsD.increment('iam_ssoe_oauth.call_to_introspect.total')
+      iam_profile = build_profile(@access_token)
 
       user_identity = build_identity(iam_profile)
+      StatsD.increment('iam_ssoe_oauth.create_user_profile.success',
+                       tags: ["credential:#{iam_profile[:fediamauth_n_type]}"])
+      Rails.logger.info('IAMUser create user session: user profile built')
       session = build_session(@access_token, user_identity)
       user = build_user(user_identity)
       handle_nil_user(user_identity) if user.nil?
       validate_user(user)
       create_evss_account(user) if user.authorize(:evss, :access?)
       log_session_info(iam_profile, user_identity, @access_token)
-      persist(session, user)
+      user = persist(session, user)
+      StatsD.increment('iam_ssoe_oauth.user_session_creation_done')
+      user
     rescue Common::Exceptions::Unauthorized => e
       Rails.logger.error('IAMUser create user session: unauthorized', error: e.message)
       StatsD.increment('iam_ssoe_oauth.inactive_session')
       raise e
     end
 
+    def build_profile(access_token)
+      iam_profile = iam_ssoe_service.post_introspect(access_token)
+
+      if iam_profile
```

Comment:
As we discussed before, this guard doesn't do anything.

Reply:
its now removed